### ISSUE TITLE
active_cover not in simulation (for real tagging)

### ISSUE THREAD

**Issue by pommedeterresautee** [JohnLangford/vowpal_wabbit#1277]
The option active_cover seems to work only in simulation mode.
How can I use it with the daemon mode to effectively tag documents?

**Comment by JohnLangford** [comment]
It does not seem to exist.  See here: https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/active_cover.cc#L128 where the label is used directly. 

@alekh have you thought about making active cover work in real mode instead of simulation mode?

**Comment by pommedeterresautee** [comment]
For what it worth, on my highly skewed NLP dataset, the --active_cover query 3 times less than the --active option and provide better accuracy. 
It would be a waste to not be able to use it when the work is done. 
If nothing is planed, are you aware of a similar implementation. 

Thanks for any information you can provide.

**Comment by JohnLangford** [comment]
@alekh and I discussed.  The plan is to create a separate reduction in VW that handles the simulation mode similar to the way that --cbify works for contextual bandits.  This is a matter of doing so if anyone wants to take it on, that's of course welcome.

**Comment by JohnLangford** [comment]
@alekh we need to figure out how/when to do this.

**Comment by alekh** [comment]
Yes. First week of April?

Alekh
________________________________

**Comment by JohnLangford** [comment]
Works for me.

**Comment by JohnLangford** [comment]
@alekh have you settled enough to think about this?

**Comment by alekh** [comment]
Yes. I need to swap this back in, but I think last time when we looked at things, we realized that there was something broken in the code for non simulation. Might be a good idea to also do this for coal. Maybe Akshay and I can do both together?

Alekh
________________________________

**Comment by matanster** [comment]
[Related SO](https://datascience.stackexchange.com/questions/26391/contextual-bandits-for-online-learning/35907#35907)

**Comment by JohnLangford** [comment]
@alekh your thoughts here?

**Comment by alekh** [comment]
Sorry, I am not sure I have the bandwidth in the near future. After the ICML deadline, I really want to focus on our book.

Alekh

**Comment by JohnLangford** [comment]
Closing this, but a patch is welcome.
